src/tools/r2rdump/R2RDump.cs

```diff
@@ -4,55 +4,436 @@
+        /// <param name="exact">Specifies whether to look for methods with names/signatures/ids matching the method exactly or partially</param>
+        private void QueryMethod(R2RReader r2r, string title, IReadOnlyList<string> queries, bool exact)
+        {
+            if (queries.Count > 0)
+            {
+                WriteDivider(title);
+            }
+            foreach (string q in queries)
+            {
+                List<R2RMethod> res = new List<R2RMethod>();
+                FindMethod(r2r, q, exact, res);
+
+                Console.WriteLine(res.Count + " result(s) for \"" + q + "\"");
+                Console.WriteLine();
+                foreach (R2RMethod method in res)
+                {
+                    DumpMethod(r2r, method);
+                }
+            }
+        }
+
+        // <summary>
+        /// For each query in the list of queries, search for all sections by the name or value of the ReadyToRunSectionType enum
+        /// </summary>
+        /// <param name="r2r">Contains all the extracted info about the ReadyToRun image</param>
+        /// <param name="queries">The names/values to search for</param>
+        private void QuerySection(R2RReader r2r, IReadOnlyList<string> queries)
+        {
+            if (queries.Count > 0)
+            {
+                WriteDivider("R2R Section");
+            }
+            foreach (string q in queries)
+            {
+                List<R2RSection> res = new List<R2RSection>();
+                FindSection(r2r, q, res);
+
+                Console.WriteLine(res.Count + " result(s) for \"" + q + "\"");
+                Console.WriteLine();
+                foreach (R2RSection section in res)
+                {
+                    DumpSection(r2r, section);
+                }
+            }
+        }
+
+        // <summary>
+        /// For each query in the list of queries, search for a runtime function by id. 
+        /// The method containing the runtime function gets outputted, along with the single runtime function that was searched
+        /// </summary>
+        /// <param name="r2r">Contains all the extracted info about the ReadyToRun image</param>
+        /// <param name="queries">The ids to search for</param>
+        private void QueryRuntimeFunction(R2RReader r2r, IReadOnlyList<int> queries)
+        {
+            if (queries.Count > 0)
+            {
+                WriteDivider("Runtime Functions");
+            }
+            foreach (int q in queries)
+            {
+                Console.WriteLine("id: " + q);
+                Console.WriteLine();
+
+                R2RMethod method = null;
+                RuntimeFunction rtf = FindRuntimeFunction(r2r, q, out method);
+
+                if (method == null)
+                {
+                    WriteWarning("Unable to find by id " + q);
+                    continue;
+                }
+                DumpMethod(r2r, method, rtf);
```

Comment:
It seems more intuitive to me if there was a separate DumpRuntimeFunction call, which could be called here instead of DumpMethod. Then in the normal DumpMethod call, it would iterate to through each RuntimeFunction and call DumpRuntimeFunction. This would also simplify your FindRuntimeFunction call so you don't have the additional `out method`.